osutil/group.go

```diff
@@ -20,35 +20,112 @@
 package osutil
 
 import (
+	"bytes"
+	"fmt"
+	"os/exec"
 	"os/user"
 	"strconv"
 )
 
-// TODO: the builtin os/user functions only look at /etc/passwd and /etc/group
-// which is fine for our purposes today. In the future we may want to support
-// lookups in extrausers, which is configured via nsswitch.conf. Since snapd
-// does not support being built with cgo itself, when we want to support
-// extrausers here, we can convert these to do the equivalent of:
-//
-//   getent passwd <user> | cut -d : -f 3
-//   getent group <group> | cut -d : -f 3
-
-// FindUid returns the identifier of the given UNIX user name.
+// FindUid returns the identifier of the given UNIX user name with no getent
+// fallback
 func FindUid(username string) (uint64, error) {
-	user, err := user.Lookup(username)
+	return findUid(username)
+}
+
+// FindGid returns the identifier of the given UNIX group name with no getent
+// fallback
+func FindGid(groupname string) (uint64, error) {
+	return findGid(groupname)
+}
+
+// getent returns the identifier of the given UNIX user or group name as
+// determined by the specified database
+func getent(name string, database string) (uint64, error) {
```

Comment:
naively I would have had the two arguments in the other order to match the command